extensions/ql-vscode/src/data-extensions-editor/predicates.ts

```diff
@@ -0,0 +1,136 @@
+import { ExternalApiUsage } from "./external-api-usage";
+import {
+  ModeledMethod,
+  ModeledMethodType,
+  ModeledMethodWithSignature,
+} from "./modeled-method";
+
+export type ExternalApiUsageByType = {
+  externalApiUsage: ExternalApiUsage;
+  modeledMethod: ModeledMethod;
+};
+
+export type ExtensiblePredicateDefinition = {
+  extensiblePredicate: string;
+  generateMethodDefinition: (method: ExternalApiUsageByType) => any[];
+  readModeledMethod: (row: any[]) => ModeledMethodWithSignature;
+
+  supportedKinds?: string[];
+};
+
+function readRowToMethod(row: any[]): string {
```

Comment:
There are only specific types supported in extension files. There's an opportunity to avoid using `any`.

```suggestion
type Tuple = boolean | number | string;
function readRowToMethod(row: Tuple[]): string {
```